src/python/dxpy/scripts/dx.py

```diff
@@ -4397,6 +4409,17 @@ def register_parser(parser, subparsers_action=None, categories=('other', ), add_
 parser_list_stages.set_defaults(func=workflow_cli.list_stages)
 register_parser(parser_list_stages, subparsers_action=subparsers_list, categories='workflow')
 
+parser_list_files = subparsers_list.add_parser('files', help='List the data files associated with a database',
+                                                description='List the data files associated with a database',
+                                                parents=[env_args],
+                                                prog='dx list files')
+parser_list_files.add_argument('database', help='ID of the database.')
+parser_list_files.add_argument('--filename', help='Directory from which to start search. This will typically match the name of the table whose files are of interest. If omitted, the search starts at the root location for the database.')
```

Comment:
Are paths supported here? Like partitions and sub-partitions of a table?

Reply:
yep :). but @wkoetke  to confirm

Reply:
If so, then maybe this should not be called `filename`

Reply:
Yes, that's correct, whatever you put in for "filename" is treated a the prefix of object keys to search for. So you can put blank, or a table name, or a table+partition, etc. I also agree 'filename' is an odd name here, although as long as an arg is documented correctly then it should be ok...my thought process is naming it that was that all the TIP database file related routes (readFile, writeFile, renameFile, deleteFile, getFileStatus, getStatusList, etc.) take a 'filename' argument, where filename is always (in those other routes) a placeholder for a path that can be either a directory or a file.

Reply:
arg has been changed to "folder" per review with George